Kernel/Syscalls/alarm.cpp

```diff
@@ -14,7 +14,7 @@ KResultOr<FlatPtr> Process::sys$alarm(unsigned seconds)
     VERIFY_PROCESS_BIG_LOCK_ACQUIRED(this);
     REQUIRE_PROMISE(stdio);
     unsigned previous_alarm_remaining = 0;
-    if (m_alarm_timer) {
+    if (m_alarm_timer && m_alarm_timer->is_in_use()) {
```

Comment:
We should move this into `cancel_timer` and return `false` if it is not in use.

Reply:
We still need to distinguish between not having an active timer at all, vs having one and if we successfully cancelled it or not though?

Reply:
I just think that `cancel_timer` should internally track that, so that you can simply call `cancel_timer` on any `Timer`, whether it's being used or not. Otherwise every call site (e.g. `Thread::block`, maybe more) will have to track this as well, and this may be racy (though right now the `big_lock` in this particular instance prevents this).

Reply:
Ok, please take a look at what I came up with. I think it should address  your concerns.